bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -206,12 +272,18 @@ public ManagedChannel shutdownNow() {
   @Override
   public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
     long endTimeNanos = System.nanoTime() + unit.toNanos(timeout);
-    for (ManagedChannel channel : channels.get()) {
+    Counter counter = getActiveChannelCounter();
```

Comment:
NB: the logic here is slightly strange, although I think it's probably fine. We potentially spend our entire time budget awaiting for termination on the first non-terminated channel we find. Of course the implementation of awaitTermination doesn't actually do anything either other than just sleeping for the entire allotted time, so again I guess none of this matters, but it's still weird to me. 

It would read more natural IMO if we just first checked for any non-terminated channels, then if we find any, slept for some increment, checking all of them again, but I imagine this is not worth worrying about.